gridded_etl_tools/utils/extractor.py

```diff
@@ -311,11 +298,11 @@ def request(
         log.info(f"Downloading remote file {source} to {output}")
         with open(output, "wb") as fp:
             try:
-                # Use a semaphore to limit the number of simultaneous downloads to 1 even in multithreaded
-                # environments. This is either a requirement of ftplib or a common requirement of FTP servers.
-                with self.semaphore:
-                    log.info("Using a single thread as a requirement of FTP even if multiple threads are available")
-                    self.ftp.retrbinary(f"RETR {source}", fp.write)
+                # need separate FTP for each download to take advantage of concurrency
```

Comment:
so will these actually run concurrently?

Reply:
They will if you set the `concurrency_limit` to be more than 1. It's unclear if that's advisable for FTPs though, since they seem to be finicky with any sort of parallelism.

Reply:
In other words, the code fundamentally works, but might be slower or less reliable than just running the code linearly.

Reply:
I think I tried multiple connections and it was either slower or didn't work. I was mostly looking at [this thread](https://stackoverflow.com/questions/2549829/using-ftplib-for-multithread-uploads) at the time. If concurrency is 1 by default, then maybe it's fine to leave it as an option though.